common/src/main/java/org/apache/uniffle/common/netty/protocol/GetLocalShuffleIndexResponse.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.netty.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+import org.apache.uniffle.common.rpc.StatusCode;
+import org.apache.uniffle.common.util.ByteBufUtils;
+
+public class GetLocalShuffleIndexResponse extends RpcResponse {
+  private ByteBuf indexData;
+  private long fileLength;
+
+  public GetLocalShuffleIndexResponse(long requestId, StatusCode statusCode, byte[] indexData, long fileLength) {
+    this(requestId, statusCode, null, indexData, fileLength);
+  }
+
+  public GetLocalShuffleIndexResponse(long requestId, StatusCode statusCode, String retMessage,
+      byte[] indexData, long fileLength) {
+    this(requestId, statusCode, retMessage, Unpooled.wrappedBuffer(indexData), fileLength);
+  }
+
+  public GetLocalShuffleIndexResponse(long requestId, StatusCode statusCode, String retMessage,
+      ByteBuf indexData, long fileLength) {
+    super(requestId, statusCode, retMessage);
+    this.indexData = indexData;
+    this.fileLength = fileLength;
+  }
+
+  @Override
+  public int encodedLength() {
+    return super.encodedLength() + Integer.BYTES + indexData.readableBytes() + Long.BYTES;
+  }
+
+  @Override
+  public void encode(ByteBuf buf) {
+    super.encode(buf);
+    ByteBufUtils.copyByteBuf(indexData, buf);
+    buf.writeLong(fileLength);
+  }
+
+  public static GetLocalShuffleIndexResponse decode(ByteBuf byteBuf) {
```

Comment:
If I understand correctly, this method is for client to decode into `GetLocalShuffleIndexResponse` from `byteBuf`.

Can we have a pluggable decoder/encoder to unify different responses and requests? 

The `decode` also should be as an interface method ?

Reply:
Spark does the same, See https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/protocol/RpcResponse.java.

Reply:
I prefer this way.